### PR TITLE
fix(workflow): Actually use 1000 limit in bulk query action

### LINKED ISSUE #6099: Bulk merge doesn't apply to all selected
Steps to reproduce
- Use the "Select the first 1,000 issues that match this search query" feature to select multiple pages of issues
- Use the UI to bulk merge issues

Upon refresh, some of the issues appear to have been merged, but many are not merged.

I did wonder if this was because the work happens asynchronously, however some issues appear to be merged immediately, leaving the user to believe the work has already completed.

### PR DESCRIPTION
**Pull request by MaxBittker**
Ready to merge, fixes https://github.com/getsentry/sentry/issues/6099

### CODE REVIEW

File: src/sentry/search/django/backend.py [hunk 1]
```diff
@@ -273,7 +273,7 @@ def _build_queryset(
         )
         return queryset
 
-    def query(self, project, count_hits=False, **kwargs):
+    def query(self, project, count_hits=False, paginator_options={}, **kwargs):
```
Comment (tkaemming) [inline src/sentry/search/django/backend.py]: Don't get into the bad habit of using mutable function arguments: http://docs.python-guide.org/en/latest/writing/gotchas/#mutable-default-arguments

File: src/sentry/api/endpoints/project_group_index.py [hunk 2]
```diff
@@ -450,7 +450,9 @@ def put(self, request, project):
             # a bit too complicated right now
             query_kwargs['limit'] = 1000
```
Comment (tkaemming) [inline src/sentry/api/endpoints/project_group_index.py]: Maybe assign this to a local variable `limit` (e.g. `limit = query_kwargs['limit'] = 1000`) and use it below too?

**Review by tkaemming — CHANGES_REQUESTED** [review]

**Comment by MaxBittker** [comment]
this test fails when max limit is set to 100 in project_group_index:
`paginator_options={'max_limit': 100}, **query_kwargs)`

File: src/sentry/search/django/backend.py (hunk 1 above)
Comment (MaxBittker) [reply src/sentry/search/django/backend.py]: wow i had no idea ted

Comment (MaxBittker) [reply src/sentry/search/django/backend.py]: good lord

Comment (tkaemming) [reply src/sentry/search/django/backend.py]: 🙃

**Comment by MaxBittker** [comment]
for the record, the looped 200 inserts still don't make this the slowest test: 
```
================================ slowest 10 test durations =================================
3.80s setup    tests/sentry/api/endpoints/test_project_group_index.py::GroupListTest::test_auto_resolved
1.35s call     tests/sentry/api/endpoints/test_project_group_index.py::GroupUpdateTest::test_bulk_resolve
1.26s teardown tests/sentry/api/endpoints/test_project_group_index.py::GroupDeleteTest::test_global_is_forbidden
0.86s call     tests/sentry/api/endpoints/test_project_group_index.py::GroupListTest::test_auto_resolved
0.22s call     tests/sentry/api/endpoints/test_project_group_index.py::GroupListTest::test_stats_period
0.21s call     tests/sentry/api/endpoints/test_project_group_index.py::GroupUpdateTest::test_discard
0.19s call     tests/sentry/api/endpoints/test_project_group_index.py::GroupDeleteTest::test_delete_by_id
0.19s call     tests/sentry/api/endpoints/test_project_group_index.py::GroupListTest::test_lookup_by_release
```

File: tests/sentry/api/endpoints/test_project_group_index.py [hunk 3]
```diff
@@ -362,6 +362,39 @@ def test_global_resolve(self):
             group=new_group4,
         )
 
+    def test_bulk_resolve(self):
+        self.login_as(user=self.user)
+
+        for i in range(200):
+            self.create_group(checksum=six.binary_type(i) * 2, status=GroupStatus.UNRESOLVED)
```
Comment (tkaemming) [inline tests/sentry/api/endpoints/test_project_group_index.py]: I don't think you actually need to pass `checksum` at all: https://github.com/getsentry/sentry/blob/6e47bc607827da5fb8a9976e31d86d59c3607465/src/sentry/testutils/fixtures.py#L523-L525

File: src/sentry/api/endpoints/project_group_index.py [hunk 4]
```diff
@@ -441,16 +451,20 @@ def put(self, request, project):
...
 
             # bulk mutations are limited to 1000 items
             # TODO(dcramer): it'd be nice to support more than this, but its
             # a bit too complicated right now
-            query_kwargs['limit'] = 1000
+            limit = 1000
+            query_kwargs['limit'] = limit
 
-            cursor_result = search.query(**query_kwargs)
+            # the paginator has a default max_limit of 100, which must be overwritten.
```
Comment (tkaemming) [inline src/sentry/api/endpoints/project_group_index.py]: 👍

**Review by tkaemming — APPROVED** [review]
(I'm assuming the tests actually pass here.)

**Comment by tkaemming** [comment]
Oh, also fix your commit message to spec or whatever

**Comment by getsentry-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">You should update CHANGES due to the size of this PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

**Comment by MaxBittker** [comment]
codecov is mad because of the format changes :(

**Comment by kyawko41493** [comment]
I like this

**Comment by OliverJAsh** [comment]
❤️

**Pull request merged**